### PR TITLE
feat: add examples for rust, spring, c# on SN Cloud

### PR DESCRIPTION
**Pull request by ericsyh**

### CODE REVIEW

**Comment by momo-jun** [comment]
Not sure whether it's the proper place to ask - the `cloud` folder contains multiple README files and most of the content is duplicated with [SN docs](https://docs.streamnative.io/docs/connect-overview), e.g.:
* https://github.com/streamnative/examples/blob/master/cloud/README.md
* https://github.com/streamnative/examples/tree/master/cloud/java
* https://github.com/streamnative/examples/tree/master/cloud/cpp
* ...

I wonder if it makes more sense to use links to replace the duplicate content? If not, we need to double the maintenance effort while the sync is still easier to miss.
//cc @Huanli-Meng

File: cloud/spring/TokenApplication.yml
Comment (Huanli-Meng) [inline cloud/spring/TokenApplication.yml:1]: Out of curiosity, we define the Pulsar cluster info and auth info in the YAML file. How should we create the client? use the `kubectl` command to apply or perform any other action? Thanks

**Review by Huanli-Meng — APPROVED** [review]
Only one question.

File: cloud/c#/TokenConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace CsharpExamples
+{
+    internal class JWT
+    {
+        internal static async Task Consumer()
+        {
+            const string serviceUrl = "{{ SERVICE_URL }}";
+            const string subscriptionName = "test-sub";
+            var topicName = "persistent://public/default/test-topic";
+            var token = "{{ AUTH_PARAMS_TOKE }}";
```
Comment (momo-jun) [inline cloud/c#/TokenConsumer.cs:15]:
```suggestion
            var token = "{{ AUTH_PARAMS_TOKEN }}";
```
As pointed out by Huanli in [this comment](https://github.com/streamnative/sn-docs/pull/236#discussion_r1166542763) - not sure if it is intentional or a typo? There are multiple occurrences in the commit.

File: cloud/rust/TokenConsumer.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+use pulsar::{Authentication, Consumer, ConsumerOptions, Pulsar, SubType, TokioExecutor};
+use pulsar::consumer::InitialPosition;
+
+#[tokio::main]
+async fn main() -> Result<(), pulsar::Error> {
+    env_logger::init();
+
+    let addr = "{{ SERVICE_URL }}".to_string();
+    let mut builder = Pulsar::builder(addr, TokioExecutor);
+    let token = "{{ AUTH_PARAMS_TOKE }}".to_string();
```
Comment (momo-jun) [inline cloud/rust/TokenConsumer.rs:12]:
```suggestion
    let token = "{{ AUTH_PARAMS_TOKEN }}".to_string();
```
There are several more occurrences that need to be corrected too:)

File: cloud/spring/TokenApplication.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+spring:
+  pulsar:
+    client:
+      service-url: {{ SERVICE_URL }}
+      auth-plugin-class-name: org.apache.pulsar.client.impl.auth.AuthenticationToken
+      authentication:
+        token: {{ AUTH_PARAMS_TOKE }}
```
Comment (momo-jun) [inline cloud/spring/TokenApplication.yml:8]:
```suggestion
        token: {{ AUTH_PARAMS_TOKEN }}
```

**Pull request merged**